Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -532,8 +537,8 @@ private DTParseResult ParseNumberWithUnit(string text, DateObject referenceDate)
                                 endDate = referenceDate.AddDays((double)pr.Value);
                                 break;
                             case "W":
-                                beginDate = referenceDate;
-                                endDate = referenceDate.AddDays(7 * (double)pr.Value);
+                                beginDate = beforeStr.ToLower().Equals("in") ? referenceDate.AddDays(7 * ((double)pr.Value - 1)) : referenceDate;

```

Comment:
@rubio41 We should not have language-specific stuff in here. Let's move this to the culture config.